Assertion failure in `_nfloat_vec_dot_add_limbs` with -O0

```
$ ./bootstrap.sh && ./configure --enable-assert CFLAGS='-O0' && make -j32
$ make check MOD=nfloat
Makefile:518: Dependency tracking only set to cover the test executables of nfloat.
/usr/bin/mkdir -p build/nfloat/test
  CC  nfloat/test/main.c
add_sub_n...
add_sub_n                                         0.17   (PASS)
addmul_submul...
addmul_submul                                     0.00   (PASS)
complex_mat_mul...
main: /home/mezzarobba/co/flint/src/nfloat/dot.c:86: _nfloat_vec_dot_add_limbs: Assertion `delta < 64' failed.
make: *** [Makefile:776: build/nfloat/test/main_TEST_RUN] Aborted (core dumped)
```
Does not happen without `-O0`.

--- a/src/nfloat/dot.c
+++ b/src/nfloat/dot.c
@@ -83,8 +83,6 @@ _nfloat_vec_dot_add_limbs(ulong * s, slong sexp, mp_srcptr x, slong xexp, int su
         delta_bits = xexp - sexp;
         delta_limbs = 0;
 
-        FLINT_ASSERT(delta < FLINT_BITS);
-
         mpn_lshift(t, x, n, delta_bits);
 
         if (subtract)

